test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/TestAutoCreateSharedArchive.java

```diff
@@ -0,0 +1,693 @@
+ *    archive will be created at exit if base can be shared.
+ *
+ * 23  mismatched jvm_indent in base/top archive
+ *     23.01 mismatched jvm_indent in top archive
+ *     23.02 mismatched jvm_indent in base archive
+ *
+ * 24 run with non-existing shared archives
+ *   24.01 run -Xshare:auto -XX:+AutoCreateSharedArchive -XX:SharedArchiveFile=base.jsa:non-exist-top.jsa
+ *     The top archive will be regenerated.
+ *   24.02 run -Xshare:auto -XX:+AutoCreateSharedArchive -XX:SharedArchiveFile=non-exist-base.jsa:top.jsa
+ *     top archive will not be shared if base archive failed to load.
+ */
+
+import java.io.IOException;
+import java.io.File;
+
+import java.nio.file.attribute.FileTime;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import jdk.test.lib.cds.CDSTestUtils;
+import jdk.test.lib.cds.CDSArchiveUtils;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.helpers.ClassFileInstaller;
+
+public class TestAutoCreateSharedArchive extends DynamicArchiveTestBase {
+    private static final String BASE_NAME = CDSTestUtils.getOutputFileName("base.jsa");
+    private static final String TOP_NAME  = CDSTestUtils.getOutputFileName("top.jsa");
+    private static final String mainAppClass = "Hello";
+    private static final String HELLO_SOURCE = "Hello source: shared objects file (top)";
+    private static final String HELLO_WORLD = "Hello World";
+    private static boolean verifyOn = false;
+
+    private static int   genericHeaderMinVersion = CDSArchiveUtils.getGenericHeaderMinVersion();
+    private static int   currentCDSVersion = CDSArchiveUtils.getCurrentCDSArchiveVersion();
+
+    public static void main(String[] args) throws Exception {
+        if (args.length != 1 || (!args[0].equals("verifySharedSpacesOff") && !args[0].equals("verifySharedSpacesOn"))) {
+            throw new RuntimeException("Must run with verifySharedSpacesOff or verifySharedSpacesOn");
+        }
+        verifyOn = args[0].equals("verifySharedSpacesOn");
+        runTest(TestAutoCreateSharedArchive::testAutoCreateSharedArchive);
+    }
+
+    public static void checkFileExists(String fileName) throws Exception {
+        File file = new File(fileName);
+        if (!file.exists()) {
+             throw new IOException("Archive " + fileName + " is not automatically created");
+        }
+    }
+
+    public static String startNewArchive(String testName) {
+        String newArchiveName = TestCommon.getNewArchiveName(testName);
+        TestCommon.setCurrentArchiveName(newArchiveName);
+        return newArchiveName;
+    }
+
+    public static void print(String message) {
+        System.out.println(message);
+    }
+
+    private static void testAutoCreateSharedArchive() throws Exception {
+        String appJar = ClassFileInstaller.getJarPath("hello.jar");
+        boolean fileModified = false;
+
+        String VerifySharedSpaces = verifyOn ? "-XX:+VerifySharedSpaces" : "-XX:-VerifySharedSpaces";
```

Comment:
A local variable name usually starts with lowercase.
Suggest changing VerifySharedSpaces to verifySharedSpaces.